pkg/cloudprovider/plugins/azure/azure_compute.go

```diff
@@ -112,20 +125,43 @@ func (computeCfg *computeServiceConfig) waitForInventoryInit(duration time.Durat
 	return backoff.Retry(operation, b)
 }
 
-// getCachedVirtualMachines returns virtualMachines from the cache for the subscription.
-func (computeCfg *computeServiceConfig) getCachedVirtualMachines() []*virtualMachineTable {
+// getCachedVirtualMachines returns virtualMachines specific to a selector from the cache for the subscription.
+func (computeCfg *computeServiceConfig) getCachedVirtualMachines(selector *types.NamespacedName) []*virtualMachineTable {
 	snapshot := computeCfg.resourcesCache.GetSnapshot()
 	if snapshot == nil {
-		azurePluginLogger().V(4).Info("Compute service cache snapshot nil", "type", providerType, "account", computeCfg.account)
+		azurePluginLogger().V(4).Info("cache snapshot nil",
+			"type", providerType, "account", computeCfg.accountNamespacedName)
+		return []*virtualMachineTable{}
+	}
+
+	virtualMachines, found := snapshot.(*cacheSnapshot).vms[*selector]
+	if !found {
+		azurePluginLogger().V(4).Info("Vm snapshot nil",
+			"type", providerType, "account", computeCfg.accountNamespacedName)
 		return []*virtualMachineTable{}
 	}
-	virtualMachines := snapshot.(*computeResourcesCacheSnapshot).virtualMachines
+
 	instancesToReturn := make([]*virtualMachineTable, 0, len(virtualMachines))
-	for _, virtualMachine := range virtualMachines {
-		instancesToReturn = append(instancesToReturn, virtualMachine)
+	instancesToReturn = append(instancesToReturn, virtualMachines...)
+	azurePluginLogger().V(1).Info("Cached vm instances", "account", computeCfg.accountNamespacedName,
```

Comment:
nit: Remove this log, since we dump the compute resources in the account manager.